sway/config.c

```diff
@@ -940,10 +951,22 @@ static void invoke_swaybar(struct bar_config *bar) {
 				NULL,
 			};
 
+			close(filedes[1]);
 			execvp(cmd[0], cmd);
 			free(command);
+			exit(EXIT_SUCCESS);
+		}
+	}
+	close(filedes[0]);
+	char *buf = malloc(1024);
```

Comment:
Can you write an int with the length, then allocate an appropriately sized buffer for it?